src/Components/CategoryFacet.tsx

```diff
@@ -0,0 +1,185 @@
+
+  private headlessCategoryFacet: CategoryFacetType;
+  state: CategoryFacetState & {
+    inputValue: "";
+  };
+
+  constructor(props: any) {
+    super(props);
+
+    this.headlessCategoryFacet = buildCategoryFacet(headlessEngine, {
+      options: {
+        numberOfValues: 3,
+        field: this.props.field,
+        delimitingCharacter: "|"
+      }
+    });
+
+    this.state = this.headlessCategoryFacet.state;
+  }
+
+  componentDidMount() {
+    this.headlessCategoryFacet.subscribe(() => this.updateState());
+  }
+
+  componentWillUnmount() {
+    this.headlessCategoryFacet.subscribe(() => {});
+  }
+
+  updateState() {
+    this.setState(this.headlessCategoryFacet.state);
+  }
+
+  toggleSelect(value: CategoryFacetValue) {
+    this.headlessCategoryFacet.toggleSelect(value);
+  }
+
+  showMore() {
+    this.headlessCategoryFacet.showMoreValues();
+  }
+
+  showLess() {
+    this.headlessCategoryFacet.showLessValues();
+  }
+  //sends left position offset value to getValues <Box> depending on state
+  parentAlign() {
+    return this.state.hasActiveValues ? "20%" : "0%";
+  }
+
+  getClearButton() {
+    if (!this.state.hasActiveValues) {
+      return null;
+    }
+    return (
+      <div onClick={() => this.headlessCategoryFacet.deselectAll()}>
+        <Typography>
+          <IconButton edge="start" size="small">
+            <ChevronLeftIcon />
+          </IconButton>
+          {this.props.subtitle}
+        </Typography>
+      </div>
+    );
+  }
+
+  getParents() {
+    const parents = this.state.parents;
+
+    return parents.map((parent, i) => {
+      const isLast = i === parents.length - 1;
+      return (
+        <div
+          onClick={() =>
+            !isLast && this.headlessCategoryFacet.toggleSelect(parent)
+          }
+        >
+          <Typography align="left" color="textSecondary" style={hoveredStyle}>
+            {!isLast && (
+              <IconButton edge="start" size="small">
+                <ChevronLeftIcon />
+              </IconButton>
+            )}
+            {parent.value}
+          </Typography>
+        </div>
+      );
+    });
+  }
+
+  getValues() {
+    //variable to recieve position offset
+    const align = this.parentAlign();
+
+    return this.state.values.map((value) => (
+      <div
+        key={value.value}
+        onClick={() => this.headlessCategoryFacet.toggleSelect(value)}
+      >
+        <Box position="relative" left={align}>
+          <Typography align="left" style={hoveredStyle}>
+            {value.value} ({value.numberOfResults})
+          </Typography>
+        </Box>
+      </div>
+    ));
+  }
+
+
+  getShowMore() {
+    return (
+      <Button
+        onClick={() => {
+          this.showMore();
+        }}
+      >
+        Show More
+      </Button>
+    );
+  }
+
+  getShowLess() {
+    return (
+      <Button
+        onClick={() => {
+          this.showLess();
+        }}
+      >
+        Show Less
+      </Button>
+    );
+  }
+
+  render() {
```

Comment:
I noticed that there's a warning in the console related to the Category Facet:

![Screen Shot 2021-02-15 at 12 20 13](https://user-images.githubusercontent.com/23503066/107977319-e4cfef80-6f88-11eb-988f-746919cbe0ca.png)

@kaitlynkatz , @jpmarceau rings a bell?

Reply:
Oh yeah, this happens all the time 🙃 
We just need to provide keys on list items or React is very very sad. In this case, we have to add it at like 108, I'll comment there 👍 

(I'm kinda mad at myself for not answering this question earlier btw, considering how simple it is 😅 I saw the comment in my emails but completely forgot 😭)